unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/ChatWidgetHUD/ChatHUDView.cs

```diff
@@ -452,13 +453,8 @@ private void SortEntriesImmediate()
         {
             if (this.entries.Count <= 0) return;
 
-            var entries = this.entries.Values.OrderBy(x => x.Model.timestamp).ToList();
-
-            for (var i = 0; i < entries.Count; i++)
-            {
-                if (entries[i].transform.GetSiblingIndex() != i)
-                    entries[i].transform.SetSiblingIndex(i);
-            }
+            foreach (var entry in entries.Values.OrderBy(obj => obj.Model, SortingStrategy))
+                entry.transform.SetAsLastSibling();
```

Comment:
OrderBy allocates lot of memory as well because it duplicates the collection.

The core issue that prevents this code from being optimal is that `entries` is a Dictionary. Do we need accessing by `messageId`? Because then we could have a List and apply `List.Sort` which is super efficient. https://stackoverflow.com/a/53385164

Another solution is to use a SortedDictionary https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.sorteddictionary-2?view=net-7.0. It allocates more memory than the SortedList but both would be better than the current implementation.

Reply:
We need to index them by id because sometimes the message arrives duplicated. Depending if its already added we do an update or an addition to the list. I will try with the `SortedDictionary` and let you know the results

Reply:
I tried with a `SortedDictionary` but it only allows to sort by key (string), which is not useful for this case, because the sorting depends on the message information, not the id

Reply:
yeah, let's try to figure this out later.

Reply:
If your only use case for the dictionary is to avoid duplicating entries, you can always use an additional Hashset to lookup. Anyway, approving.

Reply:
We need to update the message information if its duplicated. `HashSet` will throw an exception when adding a duplicate. However, its better than a `List`